### ISSUE TITLE
how to setup?

### ISSUE THREAD

**Issue by sandrock** [xingheng/taskwarrior-data#1]
I am trying to use your project.  I do not understand the 2 last steps.

I cloned your repo:

```
sandrock@firefly-mjr  ~  cd .opt
sandrock@firefly-mjr  ~/.opt  git clone
sandrock@firefly-mjr  ~/.opt  git clone https://github.com/xingheng/taskwarrior-data.git --branch dev
Cloning into 'taskwarrior-data'...
remote: Enumerating objects: 174, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 174 (delta 4), reused 2 (delta 2), pack-reused 167
Receiving objects: 100% (174/174), 26.00 KiB | 3.25 MiB/s, done.
Resolving deltas: 100% (120/120), done.
sandrock@firefly-mjr  ~/.opt  cd taskwarrior-data
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   ls
data  git-sync  git-sync-on-inotify  hooks  README.md
```

I created a repo somewhere to host data and did the commands:

```
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   git remote add mine git@xxxxx.xxx:xxx/xxx.git
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   git fetch mine
warning: no common commits
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), 262 bytes | 262.00 KiB/s, done.
From xxxxxx.xxx:xxx/xxx
 * [new branch]      master     -> mine/master
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   git checkout --track mine/master
branch 'master' set up to track 'mine/master'.
Switched to a new branch 'master'
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   git config --bool branch.master.sync true
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   git config --bool branch.master.syncNewFiles true
sandrock@firefly-mjr  ~/.opt/taskwarrior-data   ls
README.md
```

The checkout commands makes the files from your remote not available. I think it will not work this way. What should I do?

**Comment by xingheng** [comment]
> Sorry for this late response, I missed it from my mailing list.

#### TLDR:
You should create a main branch (e.g. master) based on the `origin.dev` branch first, then push it to your private remote repo.

#### Explain:
For your above steps, the local `master` branch tracked your `mine`'s branch, since you used the `git checkout --track mine/master` to create the master branch, you must rebase it onto the `origin/dev` branch: `git rebase origin/dev master`, then push it your `mine` remote manually, with `-f` if necessary.

**Comment by xingheng** [comment]
Closed as inactive.
